presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java

```diff
@@ -626,6 +635,9 @@ StatementClient startQuery(String sql, Map<String, String> sessionPropertiesOver
         Map<String, String> allProperties = new HashMap<>(sessionProperties);
         allProperties.putAll(sessionPropertiesOverride);
 
+        int millis = networkTimeoutMillis.get();
+        Duration timeout = (millis > 0) ? new Duration(millis, MILLISECONDS) : new Duration(1, DAYS);
```

Comment:
Why 1 day? 
Also, did you consider doing conversion from `milliseconds == 0` to 1 day immediately in `setNetworkTimeout`? I know this would make it visible in `getNetworkTimeout()`, but maybe it's a good thing? I didn't find anything in  `getNetworkTimeout()`'s preventing this.

Reply:
I'll add a comment. I just picked 1 day as a huge value as 0 milliseconds means no timeout. I could increase it to say 999 days to be even more clear. We want `getNetworkTimeout()` to return the same value (0) that was passed to `setNetworkTimeout()` (the docs say *"zero means there is no limit"*).